ghz 0.52.0 (new formula)

bojand opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

SMillerDev commented on Formula/ghz.rb:
```diff
@@ -0,0 +1,21 @@
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/ghz -v 2>&1")
```
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

SMillerDev commented on Formula/ghz.rb:
```diff
@@ -0,0 +1,21 @@
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
```
This shouldn't be needed

SMillerDev commented on Formula/ghz.rb:
```diff
@@ -0,0 +1,21 @@
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+    system "go", "build",
```
wouldn't `std_go_args` be enough?

bayandin commented:
Should we merge https://github.com/Homebrew/homebrew-core/pull/54220 into this PR?

chenrui333 commented:
> Should #54220 be merged into this PR?

That is really good point. 
@bojand, do you think folks need to install separate executables?

bojand commented:
Hello and thanks for the input; I will work on addressing the feedback from above. Regarding #54220 and the two separate formula, I was unsure about it but in the end decided to do two separate formulas. A bit more background... `ghz-web` is an optional complementary server and web app for `ghz` CLI. People installing `ghz` may not necessarily use or want to install `ghz-web`. I would hazard a guess that most people use only `ghz`, but I really do not have any kind of usage data for this, just a hunch. Both projects' source code are in same repository, but in essence they are separate tools with different installation and run time requirements. One is a CLI tool, the other a web server component. I do not feel strongly about it one way or another. This is my first brew contribution to address [ghz#172](https://github.com/bojand/ghz/issues/172), so if there are established patterns regarding packaging associated (but not coupled) tooling similar to this context I'll follow whatever is recommended.

chenrui333 commented:
> Hello and thanks for the input; I will work on addressing the feedback from above. Regarding #54220 and the two separate formula, I was unsure about it but in the end decided to do two separate formulas. A bit more background... `ghz-web` is an optional complementary server and web app for `ghz` CLI. People installing `ghz` may not necessarily use or want to install `ghz-web`. I would hazard a guess that most people use only `ghz`, but I really do not have any kind of usage data for this, just a hunch. Both projects' source code are in same repository, but in essence they are separate tools with different installation and run time requirements. One is a CLI tool, the other a web server component. I do not feel strongly about it one way or another. This is my first brew contribution to address [ghz#172](https://github.com/bojand/ghz/issues/172), so if there are established patterns regarding packaging associated (but not coupled) tooling similar to this context I'll follow whatever is recommended.

@bojand thanks for the write-up. In this sense, it makes sense to keep two formula and help with the flexible choices. 

If you can improve the tests for both formula, we are happy to accept these two requests. Thanks!!

bojand commented:
A gentle bump... Hello, is this ready to merge? Thanks.

chenrui333 commented:
@bojand Thanks for the reminder and appreciate your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!
